internal/xds/matcher/matcher_header.go

```diff
@@ -203,6 +203,35 @@ func (hsm *HeaderSuffixMatcher) String() string {
 	return fmt.Sprintf("headerSuffix:%v:%v", hsm.key, hsm.suffix)
 }
 
+// HeaderContainsMatcher matches on whether the header value contains the
+// value passed into this struct.
+type HeaderContainsMatcher struct {
+	key      string
+	contains string
+}
+
+// NewHeaderContainsMatcher returns a new HeaderContainsMatcher. The key
```

Comment:
Other languages use things like `@param` to refer to parameters. In Go docs, you can refer to the parameter with just the name of it. And since we do not error out if an empty `contains` is passed in, we cannot technically say that "empty contains string is not allowed". What do you think about writing it as follows:

```
// NewHeaderContainsMatcher returns a new HeaderContainsMatcher. key is the HTTP
// header key to match on, and contains is the value that the header should contain for 
// a successful match. An empty contains string does not work, use HeaderPresentMatcher
// in that case.
```